benchmark-tools/README.md

```diff
@@ -0,0 +1,138 @@
+In the future we will want to provide the ability for users to execute these benchmarks on a wider set of infrastructure. As Dolt becomes more mature it will become increasingly necessary to have finer grained benchmarking. For now this should suffice as a tool for contributors to identify the efficacy of their changes.
+
+## Components
+We briefly outline the components used. The main entry points are `run_benchmarks.sh` which takes care of building and running the benchmarks, and `push_output_to_dolthub.py`, which publishes benchmarking results to DoltHub. We also discuss `sysbench_scripts`, which is a way of exposing the `sysbench` scripting API.
+
+### `run_benchmarks.sh`
+This is the top level entry point for these tools. Suppose we want to compare our local, potentially dirty, copy of Dolt to some commit, say commit `19f9e571d033374ceae2ad5c277b9cfe905cdd66`. We would then run the following:
+```
+$ ./run_benchmarks.sh bulk_insert oscarbatori 19f9e571d033374ceae2ad5c277b9cfe905cdd66
+```
+
+This will tell `run_benchmarks.sh` to take the following actions:
+- build a copy of Dolt at the current commit (noting in the binary name if it is dirty)
+- clone Dolt, checkout the specified commit, and build a binary
+- run the specified `sysbench` tests with both binaries
+
+If the result of `$(git rev-parse HEAD)` is `c84f34075e55a8e3322f584986ecf808076b309c`, and the repo is dirty then we will get the following:
+```
+output/c84f34075e55a8e3322f584986ecf808076b309c.csv
+output/19f9e571d033374ceae2ad5c277b9cfe905cdd66.csv
+```
+
+Each one of these files will contain a line for each test we specified.
+
+### Requirements
+To run this stack a few things are required:
+- a bash shell to execute `run_benchmarks.sh`
+- Docker: the binaries are built inside a container
+- Python: if you want to upload results to DoltHub, we use Doltpy's tools for pushing data to DoltHub
+
+### Uploading to DoltHub
+We can upload the results to DoltHub using `push_results_to_dolthub.py` as follows:
+```
+python push_outputp_to_dolthub.py --result-directory output --remote-results-db dolthub/dolt-benchmarks-test --branch test-run
+```
+
+These results will then be available to the team for analysis, and via our API for rendering on our benchmarking documentation.
+
+### `sysbench_scripts`
+This directory contains Lua scripts that implement custom `sysbench` behavior. Above we described `sysbench` as "scriptable". In practice this means defining `sysbench` tests in Lua scripts and passing those scripts to the `sysbench` command. As an example to run `sysbench` using a predefined test one might run:
+```
+sysbench \
+  bulk_insert \
+  --table-size=1000000 \
+  --db-driver=mysql \
+  --mysql-db=test_db \
+  --mysql-user=root \
+  --mysql-host=127.0.0.1 \
+run
+```
+
+Suppose we wanted to define custom behavior, or some custom output method, then we would define `sysbench_scripts/my_custom_test.lua`, and then run:
+```
+sysbench \
+  sysbench_scripts/lua/my_custom_test.lua \
```

Comment:
```suggestion
  sysbench_scripts/my_custom_test.lua \
```

Reply:
I think we in fact want to separate `lua` from the scripts directory since we may want to define shell wrappers around the sysbench scripts once we retire the Python code for managing the output.